website/client-old/js/directives/task-list.directive.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+(function(){
+  angular
+    .module('habitrpg')
+    .directive('taskList', taskList);
+
+  taskList.$inject = [
+    '$state',
+    'User',
+    '$rootScope',
+  ];
+
+  function taskList($state, User, $rootScope) {
+    return {
+      restrict: 'EA',
+      templateUrl: 'templates/task-list.html',
+      transclude: true,
+      scope: true,
+      // scope: {
+      //  taskList: '=list',
+      //  list: '=listDetails',
+      //  obj: '=object',
+      //  user: "=",
+      // },
+      link: function($scope, element, attrs) {
+        // @TODO: The use of scope with tasks is incorrect. We need to fix all task ctrls to use directives/services
+        // $scope.obj = {};
+        function setObj (obj, force) {
+          if (!force && ($scope.obj || scope.obj !== {} || !obj)) return;
+          $scope.obj = obj;
+          setUpGroupedList();
+          setUpTaskWatch();
+        }
+
+        $rootScope.$on('obj-updated', function (event, obj) {
+          setObj(obj, true);
+        });
+
+        function setUpGroupedList () {
+          if (!$scope.obj) return;
+          $scope.groupedList = {};
+          ['habit', 'daily', 'todo', 'reward'].forEach(function (listType) {
+            groupTasksByChallenge($scope.obj[listType + 's'], listType);
+          });
+        }
+        setUpGroupedList();
+
+        function groupTasksByChallenge (taskList, type) {
+          $scope.groupedList[type] = _.groupBy(taskList, 'challenge.shortName');

```

Comment:
Heads up, we're migrating to lodash v4 (https://github.com/HabitRPG/habitica/pull/8495) which includes many breaking changes (https://github.com/lodash/lodash/wiki/Changelog#v400). Before merging this PR should be checked and updated to make sure all lodash methods are compatible with v4.